functions/initLease/src/index.ts

```diff
@@ -0,0 +1,193 @@
+import { APIGatewayEvent, Handler } from 'aws-lambda';
+import { DynamoDBClient } from "@aws-sdk/client-dynamodb";
+import { DynamoDBDocumentClient, QueryCommand, PutCommand, UpdateCommand } from '@aws-sdk/lib-dynamodb';
+import { v4 as uuidv4 } from 'uuid';
+import { StatusCodes, ApiResponse, getEnv, Stage, MaxfriseErrorCodes } from '../../common';
+import { PROPERTY_STATUS, InitLeaseRequest, LandLord, Tenant, Responsebody } from "./types"
+
+const client = new DynamoDBClient({ region: "us-west-2" });
+const ddb = DynamoDBDocumentClient.from(client);
+
+export const handler: Handler<APIGatewayEvent, ApiResponse<InitLeaseRequest>> = async (event) => {
+  if (!event.body) return new ApiResponse<Responsebody>(StatusCodes.badRequest, { message: MaxfriseErrorCodes.missingInfo.message })
+  const body = JSON.parse(event.body) as InitLeaseRequest
+  const environment = getEnv(event.requestContext.stage)
+  const housesTableName = environment === Stage.PROD ? 'houses-prod' : 'houses'
+  const paymentJobsTableName = environment === Stage.PROD ? 'paymentJobs-prod' : 'paymentJobs'
+  let isUserOwner = false
+  let houseAvailable = false
+  let jobsCreated
+
+  try {
+    const result = await validateHouse(body.user, body.houseid, housesTableName)
```

Comment:
Actually, now that I think twice about this, I don't think we should be user the user from the request, instead I believe it would be safer to use like the token or what is generated at login, so we get the user that is logged and then check validation against the logged user, what do you think?

Reply:
We don't need to block this PR due to this, but something to look at 👍🏼

Reply:
yeah we need to understand a bit better how the authorization and authentication flow will work before making a final call, but from the quick thoughts I had around this, since the `houseID` is part of the request that can cause authorized requests access data from different users, that should not be the case if somehow we get that information from the authorization token some how